apex/contrib/test/optimizers/test_adam.py

```diff
@@ -0,0 +1,188 @@
+import copy
+import math
+import random
+import unittest
+
+import torch
+import torch.nn.functional as F
+from torch import nn
+
+try:
+    import apex
+except ImportError as e:
+    HAS_APEX = False
+else:
+    HAS_APEX = True
+
+
+class Model(torch.nn.Module):
+    def __init__(self):
+        super(Model, self).__init__()
+        self.conv1 = nn.Conv2d(1, 6, 5)
+        self.relu1 = nn.ReLU()
+        self.pool1 = nn.MaxPool2d(2)
+        self.conv2 = nn.Conv2d(6, 16, 5)
+        self.relu2 = nn.ReLU()
+        self.pool2 = nn.MaxPool2d(2)
+        self.fc1 = nn.Linear(256, 120)
+        self.relu3 = nn.ReLU()
+        self.fc2 = nn.Linear(120, 84)
+        self.relu4 = nn.ReLU()
+        self.fc3 = nn.Linear(84, 10)
+        self.relu5 = nn.ReLU()
+
+    def forward(self, x):
+        y = self.conv1(x)
+        y = self.relu1(y)
+        y = self.pool1(y)
+        y = self.conv2(y)
+        y = self.relu2(y)
+        y = self.pool2(y)
+        y = y.reshape(y.shape[0], -1)
+        y = self.fc1(y)
+        y = self.relu3(y)
+        y = self.fc2(y)
+        y = self.relu4(y)
+        y = self.fc3(y)
+        y = self.relu5(y)
+        return y 
+
+
+@unittest.skipIf(not HAS_APEX, "`apex` is not found.")
+class AdamTest(unittest.TestCase):
+    def setUp(self, seed=0):
+        super().setUp()
+        torch.manual_seed(seed)
+
+        self.model = Model().cuda()
+        self.model_ = Model().cuda()
+        self.model_.load_state_dict(copy.deepcopy(self.model.state_dict()))
+
+        self.lr = 0.00001
+        params = [p for p in self.model.parameters() if p.requires_grad]
+        self.optimizer = torch.optim.Adam(params, lr=self.lr)
+
+    def testGradScaler(self):
+        params_ = [p for p in self.model_.parameters() if p.requires_grad]
+        self.optimizer_ = apex.optimizers.FusedAdam(params_, lr=self.lr, capturable=False)
```

Comment:
[trivial question] why making the optimizer an attribute of this class?

Reply:
Fixed

Reply:
You are right, it is not really necessary. I modified that so the tested optimizer is _not_ an attribute of the class.